src/flake8_boolean_trap/plugins.rs

```diff
@@ -0,0 +1,75 @@
+use rustpython_ast::{Arguments, ExprKind};
+use rustpython_parser::ast::{Constant, Expr, Located};
+
+use crate::ast::types::Range;
+use crate::check_ast::Checker;
+use crate::checks::{Check, CheckKind};
+
+fn is_boolean_arg(arg: &Located<ExprKind>) -> bool {
+    matches!(
+        &arg.node,
+        ExprKind::Constant {
+            value: Constant::Bool(_),
+            ..
+        }
+    )
+}
+
+fn add_if_boolean(checker: &mut Checker, arg: &Located<ExprKind>, kind: CheckKind) {
+    if is_boolean_arg(arg) {
+        checker.add_check(Check::new(kind, Range::from_located(arg)));
+    }
+}
+
+pub fn check_positional_boolean_in_def(checker: &mut Checker, arguments: &Arguments) {
+    for arg in arguments.args.iter().chain(arguments.posonlyargs.iter()) {
+        let hint;
+        if arg.node.annotation.is_none() {
+            continue;
+        }
+
+        if let Some(expr) = &arg.node.annotation {
+            match &expr.node {
+                ExprKind::Name { id, .. } => {
+                    hint = id.to_string();
```

Comment:
maybe `to_string` could be avoided altogether. E.g. `hint` could be a boolean that stores whether the annotation is `bool` / `"bool"`.

Here we can compare `&id` to `"bool"` and in the "Constant" branch we can match on `Constant::Str` similarly.

Reply:
Yeah that'd be preferable, I think.

Like:
```rust
let hint = match &expr.node {
  ExprKind::Name { id, .. } => id == "bool",
  ExprKind::Constant { value, .. } => ..
  _ => false,
};
```

Reply:
corrected, please advise (tests passing, no lint suggestions)